app/src/main/java/com/example/nasbackup/datastore/SmbConnectionsStateManager.kt

```
@@ -0,0 +1,78 @@
+            // Add to existing set
+            val oldSet = prefs[SMB_CONNECTIONS_KEY] ?: emptySet()
+            val newSet = oldSet + serialize(credentials)
+            prefs[SMB_CONNECTIONS_KEY] = newSet
+        }
+    }
+
+    private fun parseJsonSet(jsonSet: Set<String>): List<SmbFileContext> {
+        return jsonSet.mapNotNull { json ->
+            try {
+                objectMapper.readValue<SmbFileContext>(json)
+            } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)